### PR TITLE
Retry on Tilde key

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -59,10 +63,16 @@ public class Player : OsuScreen
...
         private PauseOverlay pauseOverlay;
         private FailOverlay failOverlay;
+        private Box retryOverlay;
+
+        private SampleChannel SampleClick;
 
         [BackgroundDependencyLoader]
         private void load(AudioManager audio, BeatmapDatabase beatmaps, OsuConfigManager config)
         {
+            AlwaysPresent = true;
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: why?

Comment (EVAST9919) [inline osu.Game/Screens/Play/Player.cs]: oops, forgot to delete it, I was trying another way.

File: osu.Game/Screens/Play/Player.cs [hunk 2]
```diff
@@ -59,10 +63,16 @@ public class Player : OsuScreen
         private HudOverlay hudOverlay;
         private PauseOverlay pauseOverlay;
         private FailOverlay failOverlay;
+        private Box retryOverlay;
+
+        private SampleChannel SampleClick;
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: wrong variable case. why called SampleClick anyway? it's not a click

File: osu.Game/Screens/Play/Player.cs [hunk 3]
```diff
@@ -157,6 +167,13 @@ private void load(AudioManager audio, BeatmapDatabase beatmaps, OsuConfigManager
                 {
                     OnRetry = Restart,
                     OnQuit = Exit,
+                },
+                retryOverlay = new Box
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: no. just use `Content.FadeColour`
this is adding 100% fill overhead

File: osu.Game/Screens/Play/Player.cs [hunk 4]
```diff
@@ -331,5 +348,41 @@ protected override bool OnExiting(Screen next)
         private Bindable<bool> mouseWheelDisabled;
 
         protected override bool OnWheel(InputState state) => mouseWheelDisabled.Value && !IsPaused;
+
+        private bool isHolded;
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: is what holded? also it's held not holded.

**Comment by peppy** [comment]
call it `HotkeyRetryOverlay` maybe? 

also you can just bring back the black overlay box for now. we'll optimise it when the time comes.

File: osu.Game/Screens/Play/RetryOverlay.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            if (args.Repeat) return false;
+
+            if (args.Key == Key.Tilde)
+            {
+                keyIsHeld = true;
+                OnKeyPressed();
+
+                Delay(500).Schedule(() =>
```
Comment (peppy) [inline osu.Game/Screens/Play/RetryOverlay.cs]: use
`task = Scheduler.AddDelay(() => {}, ativate_delay);`
then just `task?.Cancel()` in `OnKeyUp`

also note the constants for time values.

**Pull request merged**